### PR TITLE
Fix System.Diagnostic.Tracing Tests Running on UWP

### PR DESCRIPTION
**Pull request by brianrob**
Various fixes to EventSource tests to enable them to run on UWP.

### CODE REVIEW

File: src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsManifestNegative.cs [hunk 1]
```diff
@@ -18,8 +18,17 @@ namespace BasicEventSourceTests
 {
     public class TestsManifestNegative
     {
+        private static void AsserExceptionStringsEqual(Func<string> expectedStrFunc, Exception ex)
+        {
+            if(!PlatformDetection.IsNetNative)
```
Comment (davidsh) [inline src/System.Diagnostics.Tracing/tests/BasicEventSourceTest/TestsManifestNegative.cs]: nit: add space after 'if'.

File: src/System.Diagnostics.Tracing/tests/Resources/System.Diagnostics.Tracing.Tests.rd.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<Directives xmlns="http://schemas.microsoft.com/netfx/2013/01/metadata">
+  <Library>
+    <!-- TraceLogging tests need to be able to create this type using reflection. --> 
+    <Type Name="BasicEventSourceTests.UserData" Dynamic="Required All" />
+  </Library>
+</Directives>
```
Comment (davidsh) [inline src/System.Diagnostics.Tracing/tests/Resources/System.Diagnostics.Tracing.Tests.rd.xml]: nit: missing newline at end of file.

File: src/System.Diagnostics.Tracing/tests/System.Diagnostics.Tracing.Tests.csproj [hunk 3]
```diff
@@ -43,5 +43,8 @@
       <Link>CommonTest\System\PlatformDetection.cs</Link>
     </Compile>
   </ItemGroup>
+  <ItemGroup>
+    <EmbeddedResource Include="Resources\$(AssemblyName).rd.xml" />
+  </ItemGroup>
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
 </Project>
```
Comment (davidsh) [inline src/System.Diagnostics.Tracing/tests/System.Diagnostics.Tracing.Tests.csproj]: nit: missing newline at end of file.

**Comment by davidsh** [comment]
Will this PR enable #20470 to be fixed?  I realize that some of our logging tests use reflection but others are just doing normal logging but yet the events don't show up.

Can you please review discussion thread in #20470 and comment on this?  Thanks.

@vancem @DavidGoll @CIPop

**Comment by brianrob** [comment]
I don't expect this to fix #20470.  It looks like #20470 is about using reflection to get the EventSource.  This PR is about fixing EventSource tests.  I have a corresponding change going into ProjectN that will fix the rest of the EventSource tests, however that change is all about making EventSource itself work - it won't change any reflection calls you have to try and get/create the EventSource.  Let me know if I've misunderstood.

**Comment by brianrob** [comment]
Thanks for the feedback @davidsh.  All fixed.

**Comment by davidsh** [comment]
>Let me know if I've misunderstood.

Right now #20470 represents the single issue tracking the fact that our logging tests fails on UWP  ILC (.NET Native).  We are trying to figure out if it is a test bug or product bug. It does seem like using reflection to try out some event source stuff will fail because .NET Native strips out reflection. So, perhaps part of our problem is the way our tests are written. But we also have a test that tries to start logging, then does some network I/o, then sees if the logs have the right events in them. Those tests fail as well.

So, can you help us by looking at the issue and our tests and guiding us to what we should expect with logging? Is this a test-able feature in .NET Native? How can we re-write our tests to make the feature test-able under .NET Native.  Or is there a real product bug where our event source usage for networking doesn't work in .NET Native?

cc: @DavidGoll @stephentoub @vancem

**Comment by brianrob** [comment]
This is waiting on a corresponding change in ProjectN.  Until that change is checked in, this should not be merged.

**Pull request merged**